Source/Core/Core/HW/EXI/EXI.cpp

```diff
@@ -102,11 +103,17 @@ u8 SlotToEXIDevice(Slot slot)
   }
 }
 
-void Init()
+void Init(const Sram* override_sram)
 {
-  if (!g_SRAM_netplay_initialized)
+  if (override_sram)
```

Comment:
Assuming I did this right, this might keep things a little more centralized?

```
SRamHelper
{
  SRamHelper()
  {
    InitSRAM(&g_SRAM, SConfig::GetInstance().m_strSRAM);
  }
  ~SRamHelper()
  {
    File::IOFile file(SConfig::GetInstance().m_strSRAM, "wb");
    file.WriteArray(&g_SRAM, 1);
  }
};

std::optional<SRamHelper> helper;
if (override_sram)
{
  g_SRAM = *override_sram;
}
else
{
  helper = std::make_optional<SRamHelper>();
}
```

Reply:
This would make sense if we have a Constructor/Destructor pair, but since we just have Init/Shutdown functions it doesn't really improve the situation much. I guess it puts the load and write closer together...

Reply:
Yeah, coming back to this, let's leave it.